### PR TITLE
update analysis state

### PR DESCRIPTION
**Pull request by acumarav**
update analysis status based on the error report presence

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java [hunk 1]
```diff
@@ -203,8 +238,9 @@ private Analysis update(Analysis analysis) throws NotExistException {
         } catch (IOException e) {
             LOGGER.warn(Constants.AnalysisMessages.CANT_REMOVE_ANALYSIS_DIR_LOG);
         }
+        final AnalysisResultStatusDTO updatedAnalysisStatus = reEvaluateAnalysisStatus(analysis.getStatus(), resultDir);
         exists.setAnalysisFolder(analysis.getAnalysisFolder());
```
Comment (YaroslavTir) [inline src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java:242]: If I am not wrong `File analysisFolder = new File(exists.getAnalysisFolder()) ` and `resultDir` is prity much the same dir. So basicly dirictory should not exisist before you run reEvaluateAnalysisStatus.

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java [hunk 2]
```diff
@@ -183,15 +190,43 @@ public Analysis saveResults(Analysis analysis, File resultDir) {
...
+            updated = update(analysis, resultDir);
         } catch (NotExistException e) {
             LOGGER.warn(ANALYSIS_IS_NOT_EXISTS_LOG, analysis.getId());
         }
 
         return updated;
     }
 
-    private Analysis update(Analysis analysis) throws NotExistException {
+    private AnalysisResultStatusDTO reEvaluateAnalysisStatus(AnalysisResultStatusDTO originalStatus, File resultDir){
```
Comment (YaroslavTir) [inline src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java:201]: can you extract evaluate status logic in a separate class?

Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java:201]: + add unit tests

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java (hunk 1 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java:242]: You are wrong, these are two different folders

Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java]: done

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java (hunk 2 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java]: done

**Pull request merged**